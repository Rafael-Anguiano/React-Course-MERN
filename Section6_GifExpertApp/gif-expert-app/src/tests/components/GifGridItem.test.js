import React from 'react';
import {shallow} from 'enzyme'
import GifGridItem from '../../components/GifGridItem';

describe('', () => {

    const title = 'title';
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('should show the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title)
    })
    
    test('should have an image with alt and url of props', () => {
        const img = wrapper.find('img');
        // expect( img.props().src ).toBe(url)
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    })

    test('should have animation animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes('animate__fadeIn') ).toBe(true)
    })
})
