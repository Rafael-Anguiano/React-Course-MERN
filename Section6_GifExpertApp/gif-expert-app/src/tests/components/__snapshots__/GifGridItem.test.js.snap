// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should show the component correctly 1`] = `
<div
  className="card animate_animated animate__fadeIn"
>
  <img
    alt="title"
    src="https://localhost/algo.jpg"
  />
  <p>
    title
  </p>
</div>
`;
